contracts/arbitrum/connectors/uniswap/v3_swap/main.sol

```diff
@@ -0,0 +1,155 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.7.6;
+pragma abicoder v2;
+
+/**
+ * @title Uniswap v3 swap.
+ * @dev Decentralized Exchange.
+ */
+
+import {TokenInterface} from "../../../common/interfaces.sol";
+import {Helpers} from "./helpers.sol";
+import {Events} from "./events.sol";
+import "./interface.sol";
+
+abstract contract UniswapResolver is Helpers, Events {
+	/**
+	 * @dev Buy Function
+	 * @notice Swap token(sellAddr) with token(buyAddr), buy token with minimum sell token
+	 * @param buyData Data input for the buy action
+	 */
+	function buy(
+		BuyInfo memory buyData
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _buyAmt = getUint(buyData.getId, buyData.buyAmt);
+		(TokenInterface _buyAddr, TokenInterface _sellAddr) = changeEthAddress(
+			buyData.buyAddr,
+			buyData.sellAddr
+		);
+		
+		{
+			uint _slippageAmt = convert18ToDec(_sellAddr.decimals(),
+				wmul(buyData.unitAmt, convertTo18(_buyAddr.decimals(), _buyAmt))
+			);
+			require(_slippageAmt >= buyData.expectedAmt, "Too much slippage");
+		}		
+
+		bool isEth = address(_sellAddr) == wethAddr;
+		convertEthToWeth(isEth, _sellAddr, uint256(-1));
+		approve(_sellAddr, address(swapRouter), uint256(-1));
+		ISwapRouter.ExactOutputSingleParams memory params;
+
+		{
+			params = ISwapRouter.ExactOutputSingleParams({
+				tokenIn: buyData.sellAddr,
+				tokenOut: buyData.buyAddr,
+				fee: buyData.fee,
+				recipient: address(this),
+				deadline: block.timestamp + 1,
+				amountOut: _buyAmt,
+				amountInMaximum: buyData.expectedAmt,
+				sqrtPriceLimitX96: 0
+			});
+		}
+
+		uint256 _sellAmt = swapRouter.exactOutputSingle(params);
```

Comment:
you need to check slippageAmt against `_sellAmt` over here.

Reply:
require(_slippageAmt >= _sellAmt, "Too much slippage");

Reply:
Done